syntax = "proto3";

import "google/protobuf/empty.proto";

package primitives.bool;

service BoolService {
  rpc Set(BoolSetRequest) returns (google.protobuf.Empty);
  rpc Get(BoolGetRequest) returns (BoolGetResponse);
  rpc Delete(BoolDeleteRequest) returns (BoolDeleteResponse);
}

message BoolSetRequest {
  string id = 1;
  bool value = 2;
  optional uint64 ttl_seconds = 3;
}

message BoolGetRequest {
  string id = 1;
}

message BoolGetResponse {
  bool value = 1;
}

message BoolDeleteRequest {
  string id = 1;
}

message BoolDeleteResponse {
  bool removed_value = 1;
}